src/riscv/lib/src/jit/builder/sequence.rs

```diff
@@ -83,11 +90,14 @@ pub struct SequenceBuilder<'jit, MC: MemoryConfig> {
     /// Parameter pointing to the sequence result
     result_param: Pointer<ExceptionCode>,
 
+    /// Parameter pointing to the compiler instance
+    compiler_param: Pointer<D>,
+
     /// Variable storing the maximum number of steps that can be executed in the sequence.
     steps_remaining: Variable,
 }
 
-impl<'jit, MC: MemoryConfig> SequenceBuilder<'jit, MC> {
+impl<'jit, D: DispatchCompiler<MC>, MC: MemoryConfig> SequenceBuilder<'jit, D, MC> {
```

Comment:
I'm not sure we actually need to thread `D` through? I think we can happily call an extern C function that doesn't require the compiler param.